### PR TITLE
Adding support for all bitwise operators

### PR DESCRIPTION
**Pull request by antoinegelloz**

### CODE REVIEW

**Review by Pikipoo — COMMENTED** [review]
Using const defined variables like:
```go
const (
AndOperator = "&"
XorOpereator = "^"
)
```

Would allow flexible management of the grammar and ensure possible changes in operators' definition would be easily implemented.

File: checker/checker.go [hunk 1]
```diff
@@ -567,7 +598,7 @@ func (v *visitor) ConditionalNode(node *ast.ConditionalNode) reflect.Type {
 	if t1 == nil && t2 == nil {
 		return nilType
 	}
-	if t1.AssignableTo(t2) {
+	if t1 != nil && t1.AssignableTo(t2) {
```
Comment (nyhu) [inline checker/checker.go]: due to previous cases t1 != nil  should never happen, why did you modify this ?

File: expr_test.go [hunk 2]
```diff
@@ -859,14 +860,150 @@ func TestExpr(t *testing.T) {
...
+		{
+			"27-2" + constants.OpBitwiseRShift + "(+4-3)+-2",
+			24,
+		},
 	}
 
 	for _, tt := range tests {
 		program, err := expr.Compile(tt.code, expr.Env(&mockEnv{}))
-		require.NoError(t, err, "compile error")
+		require.NoError(t, err, fmt.Sprintf("compile error: %s\n", err))
```
Comment (nyhu) [inline expr_test.go]: look at require.NoError comments, you'll see that fmt is not necessary here ;)

File: file/source.go [hunk 3]
```diff
@@ -74,22 +74,3 @@ func (s *Source) findLineOffset(line int) (int32, bool) {
...
-			break
-		} else {
-			line++
-		}
-	}
-	if line == 1 {
-		return line, 0
-	}
-	return line, s.lineOffsets[line-2]
-}
```
Comment (nyhu) [inline file/source.go:95]: why did this got modifyed ?

**Review by nyhu — CHANGES_REQUESTED** [review]

**Review by Pikipoo — APPROVED** [review]

File: file/source.go (hunk 3 above)
Comment (antoinegelloz) [reply file/source.go:95]: this function wasn't used anywhere in the library and not exported.

File: checker/checker.go (hunk 1 above)
Comment (antoinegelloz) [reply checker/checker.go]: indeed, my mistake !

File: expr_test.go (hunk 2 above)
Comment (antoinegelloz) [reply expr_test.go]: alright :)

**Review by nyhu — APPROVED** [review]

**Pull request merged**